tests/functional/test_receipt.py

```diff
@@ -116,12 +116,22 @@ def test_decode_logs_unspecified_abi_gets_all_logs(owner, contract_instance):
 def test_get_failed_receipt(owner, vyper_contract_instance, eth_tester_provider):
     # Setting to '5' always fails.
     transaction = vyper_contract_instance.setNumber.as_transaction(
-        5, sender=owner, gas_limit=100000
+        5, sender=owner, gas_limit=100000, _raise_on_fail=False
     )
     receipt = owner.call(transaction)
     assert receipt.failed
 
 
+def test_get_failed_receipt_raise(owner, vyper_contract_instance, eth_tester_provider):
+    # Setting to '5' always fails.
+    transaction = vyper_contract_instance.setNumber.as_transaction(
+        5, sender=owner, gas_limit=100000, _raise_on_fail=True
+    )
+
+    with pytest.raises(TransactionError):
```

Comment:
it would be ideal to have syntax like this work:

```python
with pytest.raises(TransactionError):
    vyper_contract_instance.setNumber(5, sender=owner, gas_limit=100000, raise_on_fail=True)
```

Reply:
Updated `ContractTransaction.__call__()` in 0e4d4ef3 to achieve this

cc @NotPeopling2day